java-strings/src/main/java/com/baeldung/string/WordIndexer.java

```diff
@@ -0,0 +1,46 @@
+package com.baeldung.string.searching;
+
+import java.util.ArrayList;
+
+public class WordIndexer {
+
+    public ArrayList findWord(String textString, String word) {
+        int index = -1;
+        ArrayList<Integer> indexes = new ArrayList<Integer>();
+        StringBuilder output = new StringBuilder();
+        boolean done = false;
+        String lowerCaseTextString = textString.toLowerCase();
+        String lowerCaseWord = word.toLowerCase();
+        int wordLength = word.length();
+
+        while(!done){
+            index = lowerCaseTextString.indexOf(lowerCaseWord, index + wordLength);  // Slight improvement
+            if (index != -1) {
+                indexes.add(index);
+            }else{
+                done = true;
+            }
+        }
+        return indexes;
+    }
+
+    public ArrayList findWordUpgrade(String textString, String word) {
```

Comment:
The two methods are identical. Isn't the first one supposed to be the naive approach?